upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -152,7 +156,50 @@ func (c *ApplyClusterCmd) Run() error {
 	}
 	c.channel = channel
 
-	assetBuilder := assets.NewAssetBuilder(c.Cluster.Spec.Assets)
+	stageAssetsLifecycle := fi.LifecycleSync
+	securityLifecycle := fi.LifecycleSync
+	networkLifecycle := fi.LifecycleSync
+	clusterLifecycle := fi.LifecycleSync
+
+	switch c.Phase {
+	case Phase(""):
+		// Everything ... the default
+
+		// until we implement finding assets we need to to Ignore them
+		stageAssetsLifecycle = fi.LifecycleIgnore
+	case PhaseStageAssets:
+		networkLifecycle = fi.LifecycleIgnore
+		securityLifecycle = fi.LifecycleIgnore
+		clusterLifecycle = fi.LifecycleIgnore
+
+	case PhaseNetwork:
+		stageAssetsLifecycle = fi.LifecycleIgnore
+		securityLifecycle = fi.LifecycleIgnore
+		clusterLifecycle = fi.LifecycleIgnore
+
+	case PhaseSecurity:
+		stageAssetsLifecycle = fi.LifecycleIgnore
+		networkLifecycle = fi.LifecycleIgnore
+		clusterLifecycle = fi.LifecycleIgnore
+
+	case PhaseCluster:
+		if c.TargetName == TargetDryRun {
```

Comment:
Do these need to be different?

Reply:
> Do these need to be different?

More context please

Reply:
We shouldn't need to special case TargetDryRun.

Reply:
That is previous code, not code from this PR. https://github.com/kubernetes/kops/blob/master/upup/pkg/fi/cloudup/apply_cluster.go#L496

All I did was move the code so that I can use phase